packages/compass-query-bar/src/stores/ai-query-reducer.ts

```diff
@@ -56,15 +74,17 @@ export type AIQuerySucceededAction = {
 
 export const runAIQuery = (
   userPrompt: string
-): AIQueryThunkAction<
+): QueryBarThunkAction<
   Promise<void>,
   | AIQueryStartedAction
   | AIQueryCancelledAction
   | AIQueryFailedAction
   | AIQuerySucceededAction
 > => {
   return async (dispatch, getState) => {
-    const { aiQueryAbortController: existingAbortController } = getState();
+    const {
+      aiQuery: { aiQueryAbortController: existingAbortController },
```

Comment:
Nice catch with the thunk action not using the root 👍

Reply:
That's the only bug that I noticed, but there are other issues there like side-effects in reducer, non-serializable values in the store, using abort controller in connect directly, I'm assuming this is still poc (as is this PR anyway) and so haven't touched those. Are there any plans to address those, should I open some tickets?

Reply:
Addressed offline, updated in other prs.